test/nodes/test_context_optimizer.py

```
@@ -0,0 +1,792 @@
+        assert result == text
+
+    def test_truncation_preserves_sentences(self, optimizer):
+        text = 'First sentence. Second sentence. Third sentence. Fourth sentence.'
+        # Use a budget that can fit some but not all sentences
+        full_tokens = optimizer.count_tokens(text)
+        first_tokens = optimizer.count_tokens('First sentence.')
+        if full_tokens > first_tokens:
+            # Budget that should fit at least first sentence but not all
+            result = optimizer.truncate_to_budget(text, first_tokens + 1)
+            # Result should end at a sentence boundary
+            assert result.endswith('.') or result.endswith('sentence')
```

Ruff (0.15.7)

[warning] Call `endswith` once with a `tuple`

Merge into a single `endswith` call

(PIE810)